src/daft-physical-plan/src/ops/cross_join.rs

```diff
@@ -0,0 +1,87 @@
+use std::sync::Arc;
+
+use common_display::tree::TreeDisplay;
+use daft_logical_plan::{
+    partitioning::{
+        ClusteringSpecRef, HashClusteringConfig, RangeClusteringConfig, UnknownClusteringConfig,
+    },
+    ClusteringSpec,
+};
+use serde::{Deserialize, Serialize};
+
+use crate::PhysicalPlanRef;
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub struct CrossJoin {
+    pub left: PhysicalPlanRef,
+    pub right: PhysicalPlanRef,
+
+    /// the side that is used for the outer loop is relevant for maintaining the clustering spec
+    pub left_in_outer_loop: bool,
```

Comment:
**suggestion**: I think using an enum would make this a lot more readable than using a boolean flag. _(could also reuse it for some of the other physical joins)_

something like:
```rs
// 
pub enum JoinSide {
  Left,
  Right
}

// ...
pub struct CrossJoin {
  // ...
  pub outer_loop_side: JoinSide // JoinSide::Left | JoinSide::Right
}
```

Reply:
will try that!